File: app/src/main/java/com/prey/actions/fileretrieval/FileretrievalService.java
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2015 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.actions.fileretrieval;

import android.app.IntentService;
import android.content.Intent;

import com.prey.PreyConfig;
import com.prey.PreyLogger;

public class FileretrievalService extends IntentService {

    public FileretrievalService() {
        super(PreyConfig.TAG);
    }

    public FileretrievalService(String name) {
        super(name);
    }

    @Override
    protected void onHandleIntent(Intent intent) {
        PreyLogger.d("***************onHandleIntent");
        FileretrievalController.getInstance().run(getApplicationContext());
        stopSelf();
    }

}

File: app/src/main/java/com/prey/actions/fileretrieval/kotlin/FileretrievalService.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.actions.fileretrieval.kotlin

import android.app.IntentService
import android.content.Intent
import com.prey.kotlin.PreyConfig
import com.prey.kotlin.PreyLogger

class FileretrievalService : IntentService {
    constructor() : super(PreyConfig.TAG)

    constructor(name: String?) : super(name)

    override fun onHandleIntent(intent: Intent?) {
        PreyLogger.d("***************onHandleIntent")
        FileretrievalController.getInstance().run(applicationContext)
        stopSelf()
    }
}